src/Controller/Post/Upload.php

```diff
@@ -21,7 +21,7 @@
 use Magento\Framework\Exception\RuntimeException;
 use Magento\Framework\Math\Random;
 use Magento\Framework\Stdlib\DateTime\DateTime;
-use Magento\MediaStorage\Model\File\UploaderFactory as FileUploaderFactory;
+use Magewirephp\Magewire\Model\Upload\File\TemporaryUploaderFactory as TemporaryFileUploaderFactory;
```

Comment:
Hi @mvenghaus,

I see what your doing, but can you explain the future benefit of having this new TempFileUploader object?

Reply:
@wpoortman i didn't find a way to get the original file name out of the default uploader

Reply:
Check, and why is it required to have the original file name? Just to make it a bit more clean for me.

Reply:
after uploading i only have the hash tmp name .. but i need the original file name to show the user which files has been added
livewire adds the filename with base64 to the hashname .. so i ported the 2 functions from livewire to magewire

Reply:
![image](https://user-images.githubusercontent.com/6408081/223059472-5ee8d122-efdb-4027-937a-3ebbee9bad00.png)